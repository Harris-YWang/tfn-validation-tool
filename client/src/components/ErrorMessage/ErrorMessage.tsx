import React from 'react';
import { IErrorMsgProps } from '../../types';

const ErrorMessage = ({ errorMsg }: IErrorMsgProps) =>
	errorMsg ? <div className="alert alert-warning mt-3">{errorMsg}</div> : null;

export default ErrorMessage;
