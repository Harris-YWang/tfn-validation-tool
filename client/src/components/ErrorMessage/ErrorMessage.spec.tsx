import { shallow } from 'enzyme';
import React from 'react';
import ErrorMessage from './index';

describe('ErrorMessage Component', () => {
	it('should render when there is an error message', () => {
		const errorMsg = 'Sample Error Message';
		const wrapper = shallow(<ErrorMessage errorMsg={errorMsg} />);

		expect(wrapper).toMatchSnapshot();
	});

	it('should render nothing when no error message', () => {
		const errorMsg = '';
		const wrapper = shallow(<ErrorMessage errorMsg={errorMsg} />);

		expect(wrapper.text()).toEqual('');
	});

	it('should render warning alert', () => {
		const errorMsg = 'Sample Error Message';
		const wrapper = shallow(<ErrorMessage errorMsg={errorMsg} />);

		expect(wrapper.find('div.alert-warning').length).toBe(1);
	});
});
