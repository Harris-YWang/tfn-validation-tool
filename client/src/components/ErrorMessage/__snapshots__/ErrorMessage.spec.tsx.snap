// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage Component should render when there is an error message 1`] = `ShallowWrapper {}`;
